pkg/controller/authentication/authentication_controller.go

```diff
@@ -134,6 +176,17 @@ func (r *ReconcileAuthentication) Reconcile(request reconcile.Request) (reconcil
 	r.status.OnCRFound()
 	reqLogger.V(2).Info("Loaded config", "config", authentication)
 
+	// Verify the presence and status of the installation resource.
+	if installationCR == nil {
+		log.Error(err, "Installation not found")
+		r.status.SetDegraded("Installation not found", err.Error())
+		return reconcile.Result{}, err
+	}
+	if installationCR.Status.Variant != oprv1.TigeraSecureEnterprise {
```

Comment:
I think you may as well just move this up before any of the authentication CR stuff so it's all together.